camel/storages/vectordb_storages/base.py

```diff
@@ -0,0 +1,172 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Any, Dict, List, Optional
+from uuid import uuid4
+
+
+class VectorDistance(Enum):
+    DOT = 1
+    COSINE = 2
+    EUCLIDEAN = 3
+
+
+@dataclass
+class VectorRecord:
+    r"""Encapsulates information about a vector's unique identifier and its
+    payload, which is primarily used as a data transfer object when saving
+    to vector storage.
+
+    Attributes:
+        vector (List[float]): The numerical representation of the vector.
+        id (str, optional): A unique identifier for the vector. If not
+            provided, an random uuid will be assigned.
+        payload (Optional[Dict[str, Any]], optional): Any additional metadata
+            or information related to the vector. (default: :obj:`None`)
+    """
+    vector: List[float]
+    id: str = field(default_factory=lambda: str(uuid4()))
```

Comment:
I think we cannot do that because the id should be unique for each memory record. The default factory randomly generates uuids for convinience. However, normally the vector id should be assigned by upper layer.